rotkehlchen/chain/substrate/manager.py

```diff
@@ -0,0 +1,555 @@
+
+            return result
+
+        raise RemoteError(
+            f'{manager.chain} failed to request after trying the following nodes: '
+            f'{", ".join(requested_nodes)}',
+        )
+    return wrapper
+
+
+class SubstrateManager():
+    def __init__(
+            self,
+            chain: Chain,
+            greenlet_manager: GreenletManager,
+            msg_aggregator: MessagesAggregator,
+    ) -> None:
+        """An interface to any Substrate chain supported by Rotki.
+
+        It uses Polkascan py-substrate-interface for interacting with the
+        substrate blockchains and the Subscan API as a chain explorer.
+
+        Official substrate chains documentation:
+        https://substrate.dev/rustdocs/v2.0.0/sc_service/index.html
+        https://guide.kusama.network/docs/en/kusama-index
+        https://wiki.polkadot.network/en/
+
+        Polkascan py-scale-codec:
+        https://github.com/polkascan/py-scale-codec/tree/master
+
+        Polkascan py-substrate-interface:
+        https://github.com/polkascan/py-substrate-interface
+        https://polkascan.github.io/py-substrate-interface/base.html
+
+        Postman Substrate Interface API:
+        https://documenter.getpostman.com/view/9969999/SWT5iLXH?version=latest
+
+        Subscan API documentation:
+        https://docs.api.subscan.io
+        """
+        if chain not in Chain:
+            raise AttributeError(f'Unexpected SubstrateManager chain: {chain}')
+
+        log.debug(f'Initializing substrate {chain} manager')
+        self.chain = chain
+        self.greenlet_manager = greenlet_manager
+        self.msg_aggregator = msg_aggregator
+        self.available_node_attributes_map: DictNodeNameNodeAttributes = {}
+        self.chain_properties: ChainProperties
+        self._connect()
+
+    def _check_chain_id(self, node_interface: NodeInterface) -> None:
+        """Validate a node connects to the expected chain.
+        """
+        # Check connection and chain ID
+        try:
+            chain_id = self._get_chain_id(node_interface=node_interface)
+        except RemoteError as e:
+            log.error(str(e))
+            raise
+
+        if chain_id != str(self.chain):
+            message = (
+                f'{self.chain} connected to node at endpoint: {node_interface.url}. '
+                f'Using an unexpected chain: {chain_id}.'
+            )
+            log.error(message)
+            raise RemoteError(message)
+
+    def _check_node_synchronization(self, node_interface: NodeInterface) -> BlockNumber:
+        """Check the node synchronization comparing the last block obtained via
+        the node interface against the last block obtained via Subscan API.
+        Return the last block obtained via the node interface.
+        """
+        # Last block via node interface
+        try:
+            last_block = self._get_last_block(node_interface=node_interface)
+        except RemoteError as e:
+            log.error(str(e))
```

Comment:
Why do this? If you want to log the error, especially without any additional comment just do it at the lowest point where it is raised. And here do nothing and let it bubble up.